apps/smart-contracts/token/test/PPO.test.ts

```diff
@@ -153,4 +154,109 @@ describe('=> PPO', () => {
       expect(await ppo.balanceOf(user1.address)).to.eq(MAX_UINT256)
     })
   })
+
+  describe('# burn', () => {
+    beforeEach(async () => {
+      await setupPPO()
+      await ppo.connect(owner).mint(user1.address, 10)
+    })
+
+    it('reverts if amount > balance', async () => {
+      const callerPPOBalanceBefore = await ppo.balanceOf(user1.address)
+
+      await expect(ppo.connect(user1).burn(callerPPOBalanceBefore.add(1))).to.revertedWith(
+        'ERC20: burn amount exceeds balance'
+      )
+    })
+
+    it("doesn't decrease caller balance if amount = 0", async () => {
+      const callerPPOBalanceBefore = await ppo.balanceOf(user1.address)
+
+      await ppo.connect(user1).burn(0)
+
+      expect(await ppo.balanceOf(user1.address)).to.eq(callerPPOBalanceBefore)
+    })
+
+    it('decreases caller balance if amount < balance', async () => {
+      const callerPPOBalanceBefore = await ppo.balanceOf(user1.address)
+
+      await ppo.connect(user1).burn(callerPPOBalanceBefore.sub(1))
+
+      expect(await ppo.balanceOf(user1.address)).to.eq(1)
+    })
+
+    it('decreases caller balance if amount = balance', async () => {
+      const callerPPOBalanceBefore = await ppo.balanceOf(user1.address)
+
+      await ppo.connect(user1).burn(callerPPOBalanceBefore)
+
+      expect(await ppo.balanceOf(user1.address)).to.eq(0)
+    })
+  })
+
+  describe('# burnFrom', () => {
+    beforeEach(async () => {
+      await setupPPO()
+      await ppo.connect(owner).mint(user1.address, 10)
+    })
+
+    it('reverts if burn from zero address', async () => {
+      await expect(ppo.connect(user1).burnFrom(ZERO_ADDRESS, 1)).revertedWith(
+        'ERC20: burn amount exceeds allowance'
+      )
+    })
+
+    it('reverts if caller not approved', async () => {
```

Comment:
```suggestion
    it('reverts if allowance < amount', async () => {
```
The way you setup this testcase doesn't really verify any of the starting parameters needed for this testcase. 

You should be checking that the allowance is less than the amount being burnt and that the user has a sufficient balance to be burnt.

Edit: **you can delete this testcase**, refer to the comment below for `reverts if caller approved and amount > allowance`

Reply:
nit:
```suggestion
    it('reverts if amount > allowance', async () => {
```
to match order in revert message